tests/drivers/flash/common/src/main.c

```diff
@@ -253,5 +253,94 @@ ZTEST(flash_driver, test_flash_flatten)
 	zassert_equal(i, EXPECTED_SIZE, "Expected device to be filled wth 0xaa");
 }
 
+ZTEST(flash_driver, test_flash_copy)
+{
+	int actual_result;
+	uint8_t buf[EXPECTED_SIZE];
+	const off_t off_max = (sizeof(off_t) == sizeof(int32_t)) ? INT32_MAX : INT64_MAX;
+
+	/*
+	 * Rather than explicitly testing 128+ permutations of input,
+	 * merge redundant cases:
+	 *  - src_dev or dst_dev are invalid
+	 *  - src_offset or dst_offset are invalid
+	 *  - src_offset + size or dst_offset + size overflow
+	 *  - buf is NULL
+	 *  - buf size is invalid
+	 */
+	struct arg_exp {
```

Comment:
you would need to make this packed for it to work on m0

Reply:
packed seems kind of odd in this context. I think the issue is that the stack is too small.

Reply:
Seems it might be stack after all

Reply:
Instead of creating a large array of data on the stack, I simply created an `_inner()` function, and passed arguments directly as literals, so that data should just be in .text now.